lib/mix/tasks/absinthe.schema.json.ex

```diff
@@ -47,50 +47,69 @@ defmodule Mix.Tasks.Absinthe.Schema.Json do
 
   @introspection_graphql Path.join([:code.priv_dir(:absinthe), "graphql", "introspection.graphql"])
 
+  defmodule Options do
+    defstruct filename: nil, schema: nil, json_codec: nil, pretty: false
+
+    @type t() :: %__MODULE__{
+            filename: String.t(),
+            schema: module(),
+            json_codec: module(),
+            pretty: boolean()
+          }
+  end
+
+  @doc "Main entry point to mix task"
   def run(argv) do
     Application.ensure_all_started(:absinthe)
 
     Mix.Task.run("loadpaths", argv)
     Mix.Project.compile(argv)
 
-    {opts, args, _} =
-      OptionParser.parse(argv, strict: [schema: :string, json_codec: :string, pretty: :boolean])
-
-    schema = find_schema(opts)
-    json_codec = find_json(opts)
-    filename = args |> List.first() || @default_filename
-
-    {:ok, query} = File.read(@introspection_graphql)
+    opts = parse_options(argv)
 
-    case Absinthe.run(query, schema) do
-      {:ok, result} ->
-        create_directory(Path.dirname(filename))
-        content = json_codec.module.encode!(result, json_codec.opts)
-        create_file(filename, content, force: true)
-
-      {:error, error} ->
-        raise error
+    case generate_schema(opts) do
+      {:ok, content} -> write_schema(content, opts.filename)
+      {:error, error} -> raise error
     end
   end
 
-  defp find_json(opts) do
-    json_codec_opt = Keyword.get(opts, :json_codec, @default_codec_name)
-    module = String.to_atom("Elixir." <> json_codec_opt)
-
-    %{module: module, opts: codec_opts(module, opts)}
+  @doc "Encode the Absinthe schema to a string, using the specified codec"
+  @spec generate_schema(Options.t()) :: String.t()
+  def generate_schema(%Options{
+        pretty: pretty,
+        schema: schema,
+        json_codec: json_codec
+      }) do
+    with {:ok, query} <- File.read(@introspection_graphql),
+         {:ok, result} <- Absinthe.run(query, schema),
```

Comment:
This should probably use the newer https://hexdocs.pm/absinthe/Absinthe.Schema.html#introspect/2 function

Reply:
Oh, nice!